cmd/kubeadm/app/preflight/checks.go

```diff
@@ -511,6 +511,11 @@ func RunInitMasterChecks(cfg *kubeadmapi.MasterConfiguration) error {
 		InPathCheck{executable: "touch", mandatory: false},
 	}
 
+	// First, check if we're root separately from the other preflight checks and fail fast
+	if err := RunRootCheckOnly(); err != nil {
```

Comment:
Can you move this above the Checker list?

Also needs to be replicated on the Node side...